[breezetall] ip

breezetall opened the pull request:
# Datuk

>“Datuk is Duke in Malay.” – Me :frog:

Datuk **WILL** free your mind from having to remember things you need to do . It's,

- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/breezetall/ip).
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you 😉

And it is FREE!

Features:

- [x] Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

breezetall committed:
Level 0. Rename, Greet, Exit

breezetall committed:
Level 1. Echo

breezetall committed:
Level 2. Add, List

breezetall committed:
Level 3. Mark as Done

breezetall committed:
Level 4. ToDos, Events, Deadlines

breezetall committed:
Level 5. Handle Errors

breezetall committed:
Level 6. Delete

breezetall committed:
Level 6. Fixed exception error. Removed extra try catch.

breezetall committed:
Added saving and loading

breezetall committed:
Removed unnecessary comments

breezetall committed:
Removed unnecessary comments

breezetall committed:
Merge branch 'branch-Level-7'

breezetall committed:
Deadline now only accepts yyyy-MM-dd format. Removed whitespaces from all cmd inputs.

breezetall committed:
Files save automatically every time the list changes.

breezetall committed:
Updated invalid date format text

breezetall committed:
Add save after every list change

breezetall committed:
Remove print debug line.

breezetall committed:
Change DukeException to extend Exception instead of RuntimeException.

breezetall committed:
Merge branch 'branch-Level-8'

breezetall committed:
Add UI class

breezetall committed:
Add Parser and TaskList class

breezetall committed:
Merge branch 'A-MoreOOP'

breezetall committed:
Divide classes into packages

breezetall committed:
Merge branch 'A-Packages'

breezetall committed:
Remove unused import

breezetall committed:
Merge remote-tracking branch 'origin/add-gradle-support'

breezetall committed:
Add JUnit test for parserMark method and deleteItem method

breezetall committed:
Merge branch 'branch-A-JUnit'

breezetall committed:
Update shadowJar in gradle

breezetall committed:
Add JavaDocs

breezetall committed:
Update to fit coding standards

breezetall committed:
Add find command

breezetall committed:
Merge branch 'branch-A-JavaDoc'

breezetall committed:
Merge branch 'branch-A-CodingStandard'

breezetall committed:
Merge branch 'branch-Level-9'

breezetall committed:
Update Deadline to print in MMM dd yyyy format.

breezetall committed:
Add CheckStyle

breezetall committed:
Update gradle

breezetall committed:
Update JavaDocs

breezetall committed:
Merge branch 'master' into branch-A-CheckStyle

breezetall committed:
Merge branch 'branch-A-CheckStyle'

alphajae11 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,68 @@
+
+        ui.startMsg();
+
+        boolean finish = false;
```
Considering the coding standard, is isFinished() better naming for boolean variable?

leejaegeun-sql replied:
Considering code standard, isn't isFinished() a better way of naming this boolean variable?

leejaegeun-sql commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,54 @@
+
+    @Override
+    public String toString() {
+
```
Is this line unecessary?

leejaegeun-sql reviewed (commented):
LGTM - very few minor errors made in coding standard

jamessinmaojun commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,187 @@
+     * @return a Local date Object.
+     * @throws DukeException when the string format is not a valid date.
+     */
+    public LocalDate validDate(String str) throws DukeException {
```
Should this method be renamed `validateDate` instead of `validDate`?

jamessinmaojun commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,119 @@
+     * @param cmd an array of strings containing mark/unmark command and the index to be marked.
+     * @throws DukeException when index is out of bounds.
+     */
+    public void marked(String[] cmd) throws DukeException {
```
Although I understand what this method is trying to do from the javadocs, I will not be able to tell what the method is supposed to do from the name itself. Maybe try renaming it to something more reflective of its actual purpose?

jamessinmaojun commented on src/main/java/duke/UI.java:
```diff
@@ -0,0 +1,101 @@
+        br = new BufferedReader(new InputStreamReader(System.in));
+    }
+
+    private void line() {
```
Again, I feel this method should be named `showLine` instead of `line` to reflect the purpose of this method.

jamessinmaojun commented on src/main/java/duke/UI.java:
```diff
@@ -0,0 +1,101 @@
+        System.out.println(LINE);
+    }
+
+    public void startMsg() {
```
Similarly for these methods as well. Should be `showStartMsg` and also `showByeMsg`.

jamessinmaojun commented on src/main/java/duke/task/Todo.java:
```diff
@@ -0,0 +1,43 @@
+    public Todo(String params){
+        this.desc = params;
+        isMarked = false;
+        type = "T";
```
Considering how `type` will never change to any other value, should this be declared as a constant instea?

jamessinmaojun reviewed (commented):
Good job! Remember to stick to Java coding conventions! :D

breezetall committed:
Add GUI.

breezetall committed:
Merge branch 'branch-Level-10'

breezetall committed:
Remove main from duke class

breezetall committed:
Add assert to Storage.

In the method parseText(), the if statements only checks for the
task type. It is assumed that the correct number of parameters has
been passed to the function.

Assert has been added to check that the correct number of parameters
are given.

To prevent any incorrect inputs given to the method.

breezetall committed:
Update code to match the Code Quality standards.

parseAdd() is too long and has several arrowheads.

Separated each if else condition into their own smaller functions.

Better readability and the naming is abstract which lets readers
understand the parseAdd() method easily without needing to
know the individual code.

breezetall committed:
Add javaDocs to UI class

breezetall committed:
Merge pull request #1 from breezetall/branch-A-Assertions

Add assert to Storage.

breezetall committed:
Merge branch 'master' into branch-A-CodeQuality

breezetall committed:
Merge pull request #2 from breezetall/branch-A-CodeQuality

Update code to follow Code Quality standards

breezetall committed:
Add reminder for the earliest expiring deadline task.

breezetall committed:
Merge branch 'branch-BCD-Extension'

breezetall committed:
Refactor Duke to Datuk

breezetall committed:
Refactor DukeException to DatukException

breezetall committed:
Add exit method for bye command

breezetall committed:
Add Ui image

breezetall committed:
Update README.md

breezetall committed:
Update README.md

breezetall committed:
Update README.md

breezetall committed:
Update Ui.png
